### PR TITLE
add config option to disable mention of ReviewStack in PR bodies

### PR DESCRIPTION
**Pull request by discentem**
add config option to disable mention of ReviewStack in PR bodies

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/facebook/sapling/pull/427).
* __->__ #427

### CODE REVIEW

File: eden/scm/edenscm/ext/github/pull_request_body.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ def create_pull_request_title_and_body(
     pr_numbers_and_num_commits: List[Tuple[int, int]],
     pr_numbers_index: int,
     repository: Repository,
+    skip_reviewstack_tag: False,
```
Comment (discentem) [inline eden/scm/edenscm/ext/github/pull_request_body.py:19]: Alternatively we could allow callers to provide a template but that seems much more complex. Thoughts?

Comment (quark-zju) [reply eden/scm/edenscm/ext/github/pull_request_body.py]: Sorry for the slow review. I think template is more flexible. But this looks okay too.

**Comment by discentem** [comment]
No worries! I do have a separate PR for templates. Could use some help on
it if you want to take a look

File: eden/scm/edenscm/ext/github/pull_request_body.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ def create_pull_request_title_and_body(
     pr_numbers_and_num_commits: List[Tuple[int, int]],
     pr_numbers_index: int,
     repository: Repository,
+    reviewstack: True,
```
Comment (quark-zju) [inline eden/scm/edenscm/ext/github/pull_request_body.py:19]: `True` is not a type. I updated this to:

    reviewstack: bool=True,

File: eden/scm/edenscm/ext/github/submit.py [hunk 3]
```diff
@@ -288,6 +288,7 @@ async def rewrite_pull_request_body(
         pr_numbers_and_num_commits,
         index,
         repository,
+        reviewstack=(ui.configbool("pr", "body.reviewstack"))
```
Comment (quark-zju) [inline eden/scm/edenscm/ext/github/submit.py:291]: I renamed the config to `github.pull-request-include-reviewstack`. The config section is usually the extension name. In this case it is "github".

File: eden/scm/edenscm/ext/github/pull_request_body.py [hunk 4]
```diff
@@ -55,10 +56,16 @@ def create_pull_request_title_and_body(
...
+    if reviewstack:
+        body = f"""{commit_msg}
 {_HORIZONTAL_RULE}
 Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack]({reviewstack_url}).
 {bulleted_list}
+"""
+    else:
+        body = f"""{commit_msg}
+{_HORIZONTAL_RULE}
+{bulleted_list}
```
Comment (quark-zju) [inline eden/scm/edenscm/ext/github/pull_request_body.py:68]: I think we can skip the list and line for single-commit stack. So I updated the code to do that.

**Pull request closed without merging**